packages/shopping/public/api.js

```diff
@@ -0,0 +1,102 @@
+const api = {
+  addToCart(body, successCb, errCb) {
+    const userId = localStorage.getItem('shoppyUserId');
+    const token = localStorage.getItem('shoppyToken');
+    const url = apiUrl + `/shoppingCarts/${userId}/items`;
+    $.ajax({
+      type: 'POST',
+      url: url,
+      data: JSON.stringify(body),
+      contentType: 'application/json',
+      headers: {
+        'Authorization': `Bearer ${token}`
+      },
+      success: successCb,
+      error: errCb
+    });
+  },
+
+  getShoppingCartItems(successCb, errCb) {
+    const userId = localStorage.getItem('shoppyUserId');
+    const token = localStorage.getItem('shoppyToken');
+    if (userId) {
+      const url = apiUrl + '/shoppingCarts/' + userId;
+      $.ajax({
```

Comment:
I believe all modern browsers support [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) by now, see [Browser compatibility](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#Browser_compatibility). Can you use that API instead of jQuery AJAX?

Similarly, we can use [async/await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) instead of callbacks, see [Browser compatibility](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#Browser_compatibility).

---

For longer term, it would be great to leverage OpenAPI spec provided by the server and use either a code-generated client (see https://github.com/OpenAPITools/openapi-generator) or at least a dynamic client https://github.com/swagger-api/swagger-js. This part is of scope of this pull request though.

Reply:
> For longer term, it would be great to leverage OpenAPI spec provided 

Opened this issues https://github.com/strongloop/loopback4-example-shopping/issues/479.